cdap-tms/src/main/java/io/cdap/cdap/messaging/store/leveldb/LevelDBTableFactory.java

```diff
@@ -137,13 +137,7 @@ private void upgradeTables(long now) throws IOException {
         continue;
       }
       // only message tables should be moved
-      int firstDotIdx = dirName.indexOf('.');
-      int secondDotIdx = dirName.indexOf('.', firstDotIdx + 1);
-      if (firstDotIdx < 0 || secondDotIdx < 0) {
-        continue;
-      }
-      String tableName = dirName.substring(firstDotIdx + 1, secondDotIdx);
-      if (!tableName.equals(messageTableName)) {
+      if (!dirName.contains(messageTableName)) {
```

Comment:
I think it is better to find the firstDot, then use `dirName.startsWith(messageTableName, firstDot + 1)`, since we don't allow `"."` in the namespace.